patroni/postmaster.py

```diff
@@ -138,19 +139,17 @@ def start(pgcommand, data_dir, conf, options):
         # In order to make everything portable we can't use fork&exec approach here, so  we will call
         # ourselves and pass list of arguments which must be used to start postgres.
         env = {p: os.environ[p] for p in ('PATH', 'LD_LIBRARY_PATH', 'LC_ALL', 'LANG') if p in os.environ}
-        try:
-            proc = PostmasterProcess._from_pidfile(data_dir)
-            if proc and not proc._is_postmaster_process():
-                # Upon start postmaster process performs various safety checks if there is a postmaster.pid
-                # file in the data directory. Although Patroni already detected that the running process
-                # corresponding to the postmaster.pid is not a postmaster, the new postmaster might fail
-                # to start, because it thinks that postmaster.pid is already locked.
-                # Important!!! Unlink of postmaster.pid isn't an option, because it has a lot of nasty race conditions.
-                # Luckily there is a workaround to this problem, we can pass the pid from postmaster.pid
-                # in the `PG_GRANDPARENT_PID` environment variable and postmaster will ignore it.
-                env['PG_GRANDPARENT_PID'] = str(proc.pid)
-        except psutil.NoSuchProcess:
-            pass
+        proc = PostmasterProcess.from_pidfile(data_dir)
```

Comment:
Let's imagine that pid from `postmaster.pid` is matching with the pid of one of the Patroni threads. In this case `PostmasterProcess.from_pidfile` will return `None`, because `start_time` from `postmaster.pid` doesn't match with process `create_time`. Patroni will try to start postgres, but postgres will refuse to do it, because `postmaster.pid`
 is already "locked".

The only way to solve such a problem is to pass the `pid` from `postmaster.pid` in the `PG_GRANDPARENT_PID`.

The original code was doing that. The current code will do quite an opposite. It will force the new postgres to ignore existing running postgres process...

Reply:
good catch, there is a "not is_postmaster_process" in the previous block, however, from_pidfile checks for "proc and is_postmaster_process", so the logic will be indeed reversed.

Reply:
Yes, you are correct. I'll just back out the logic change. I misread the original code due to seeing a log with the "too much difference" line repeated. Sorry about the noise.